src/main/scala/net/snowflake/spark/snowflake/SnowflakeRelation.scala

```diff
@@ -140,19 +143,46 @@ private[snowflake] case class SnowflakeRelation(
                                    statement: SnowflakeSQLStatement,
                                    resultSchema: StructType
                                  ): RDD[T] = {
-    val format: SupportedFormat =
-      if (Utils.containVariant(resultSchema)) SupportedFormat.JSON
-      else SupportedFormat.CSV
+    if (params.useCopyUnload) {
+      val format: SupportedFormat =
+        if (Utils.containVariant(resultSchema)) SupportedFormat.JSON
+        else SupportedFormat.CSV
+
+      val rdd: RDD[String] = io.readRDD(sqlContext, params, statement, format)
```

Comment:
nit: I know this is not part of the PR, but can we refactor io -> IO?

Reply:
I will file a separate Jira for the code refactor.